annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -536,33 +587,117 @@ private TranscriptConsequenceSummary getCanonicalTranscript(VariantAnnotation gn
+            List<VariantAnnotation> gnResponseList = null;
+            // Get annotations from Genome Nexus and log if server error (e.g VEP is down) 
+            try {    
+                 gnResponseList = apiClient.fetchVariantAnnotationByGenomicLocationPOST(partitionedList,
+                    isoformOverridesSource, "", Arrays.asList(this.enrichmentFields.split(",")));
+            } catch (Exception e) {
+                LOG.error("Annotation failed for ALL variants in this partition. " + e.getMessage());
+            }
+
+            // Verify annotations coming back from Genome Nexus and log annotation failures (e.g used to be 404s)
+            if (gnResponseList != null) {
+                for (VariantAnnotation gnResponse : gnResponseList) {
+                    logAnnotationProgress(++annotatedVariantsCount, totalVariantsToAnnotateCount, postIntervalSize);
+                    if (gnResponse.isSuccessfullyAnnotated()) {
+                        gnResponseVariantKeyMap.put(extractGenomicLocationAsString(gnResponse.getAnnotationSummary().getGenomicLocation()), gnResponse);
+                    } else {
+                        LOG.warn("Annotation failed for variant " + gnResponse.getVariant());
+                    }
+                }
+            }
         }
-        // create annotated records by merging the responses from gn with their corresponding MAF record
+
         List<AnnotatedRecord> annotatedRecords = new ArrayList<>();
+        // loop through the original mutationRecords (in original sort order) and
+        // create annotated records by merging the responses from gn with their corresponding MAF record
         for (MutationRecord record : mutationRecords) {
             String genomicLocationKey = extractGenomicLocationAsString(record);
             AnnotatedRecord annotatedRecord = new AnnotatedRecord(record);
             // if not a failed annotation then convert/merge the response from gn with the maf record
             VariantAnnotation gnResponse = gnResponseVariantKeyMap.get(genomicLocationKey);
             if (gnResponse == null) {
-                summaryStatistics.addFailedAnnotatedRecordDueToServer(record, "Annotation failed due to server error", isoformOverridesSource);
-            }
-            else if (!summaryStatistics.isFailedAnnotatedRecord(annotatedRecord, record, isoformOverridesSource)) {
+                // logged above when we verify GN response
+                // but still logging for error report
+                summaryStatistics.addFailedAnnotatedRecordDueToServer(record, "Genome Nexus failed to annotate", isoformOverridesSource);
+            } else {
                 annotatedRecord = convertResponseToAnnotatedRecord(gnResponseVariantKeyMap.get(genomicLocationKey), record, replace);
+                if (summaryStatistics.isFailedAnnotatedRecord(annotatedRecord, record, isoformOverridesSource)) {
```

Comment:
yes .. here is an example of where we use the predicate for its value .. not just the logging